File: 백준/Gold/17425. 약수의 합/약수의 합.py
import sys
# sys.stdin = open("input.txt", "r")

input = sys.stdin.readline

T = int(input())

MAX = 1000000

f = [0] * (MAX+1)

for i in range(1, MAX+1):
    for j in range(i, MAX+1, i):
        f[j] += i

g = [0] * (MAX+1)

for i in range(1, MAX+1):
    g[i] = g[i - 1] + f[i]

answer = []
for _ in range(T):
    n = int(input())
    answer.append(g[n])

print(*answer, sep='\n')

File: 백준/Gold/17425. 약수의 합/약수의 합.cc
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>
#include <unordered_map>

using namespace std;

#define MAX_SIZE 1000000

int main(void){
    ios::sync_with_stdio(false);
    cin.tie(0);

    // freopen("input.txt", "r", stdin);

    int T, N;
    cin >> T;
    long long G[MAX_SIZE + 1] = {0};

    for(int i=1; i<MAX_SIZE + 1; i++){
        for(int j=i; j<MAX_SIZE + 1; j+=i){
            G[j] +=i;
        }
        G[i] += G[i - 1];
    }


    for(int i=0; i<T; i++){
        cin >> N;
        cout << G[N] << '\n';
    }

    return 0;
}